### PR TITLE
Sync VSCode prettier config paths

### PR DESCRIPTION
**Pull request by aomarks**
- Creates a `.vscode/settings.json` for each sub-package to tell the VSCode Prettier plugin which ignore file to use.  By default it looks in the root of each sub-package, and finds no `.prettierignore` file. This way, format-on-save in VSCode will automatically respect our rules.

- Recommend the vscode prettier and eslint extensions.

<strike>
We want to avoid formatting and linting all Git-ignored files, _plus a few additional patterns_. However, neither Prettier [1] nor ESLint [2] support passing multiple ignore files. So, we now instead maintain separate `.prettierignore` and `.eslintignore` files, but automatically inject the contents of `.gitignore` into them.

This replaces the previous hacky approach, where I was creating a temp file before formatting/linting. This was bad because 1) it didn't work with our husky auto-format hook, and 2) it didn't work with the VSCode Prettier plugin.

[1] https://github.com/prettier/prettier/issues/8048
[2] https://github.com/eslint/eslint/issues/9794

cc @abdonrd 
<strike>

### CODE REVIEW

**Review by abdonrd — CHANGES_REQUESTED** [review]

**Comment by abdonrd** [comment]
@aomarks thinking about this... I want to propose this alternative, which is a bit more declarative and easy to maintain.

https://github.com/Polymer/lit-html/pull/1415/files

File: lit-next.code-workspace [hunk 1]
```diff
@@ -39,6 +39,13 @@
 	],
 	"settings": {
 		"typescript.tsdk": "lit-html/node_modules/typescript/lib",
-		"debug.javascript.warnOnLongPrediction": false
+		"debug.javascript.warnOnLongPrediction": false,
+		"files.insertFinalNewline": true,
+		"files.trimTrailingWhitespace": true
```
[the remark this answers is not in the document]
Comment (aomarks) [reply lit-next.code-workspace]: Removed (though I like it as a default for files that aren't auto-formatted).

File: lit-next.code-workspace [hunk 2]
```diff
@@ -39,6 +39,13 @@
...
+	},
+	"extensions": {
+		"recommendations": [
+			"esbenp.prettier-vscode",
```
[the remark this answers is not in the document]
Comment (aomarks) [reply lit-next.code-workspace]: Done.

**Comment by aomarks** [comment]
> @aomarks thinking about this... I want to propose this alternative: #1415
> Which seems a bit more declarative and easy to maintain.

Thanks, merged.

**Comment by abdonrd** [comment]
With this #1415, are the `.vscode/settings.json` files still needed?

**Comment by aomarks** [comment]
> With this #1415, are the `.vscode/settings.json` files still needed?

Yeah, the prettier extension is a bit weird and looks for a `.prettierrc` in each root of a multi-root workspace, instead of looking for the nearest `.prettierrc` on disk, like I would expect.

**Review by justinfagnani — APPROVED** [review]

**Pull request merged**